src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,104 @@
+package duke;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Handles saving and loading tasks to/from a file.
+ */
+public class Storage {
+    private final String saveLocation;
+    private final Ui ui;
+
+    public Storage(String saveLocation, Ui ui) {
+        this.saveLocation = saveLocation;
+        this.ui = ui;
+    }
+
+    /**
+     * Saves the tasks from the provided TaskList to the specified file.
+     *
+     * @param taskList The TaskList containing tasks to be saved.
+     */
+    public void saveTasks(TaskList taskList) {
+        File saveFile = new File(saveLocation);
+        saveFile.mkdirs();
+        if (saveFile.exists()) {
+            boolean deleteFileSuccess = saveFile.delete();
+            if (!deleteFileSuccess) {
+                System.out.println("Failed to delete previous save file!");
+            }
+        }
+        try {
+            boolean createNewFileSuccess = saveFile.createNewFile();
+            if (!createNewFileSuccess) {
+                System.out.println("Failed to create save file!");
+            } else {
+                BufferedWriter writer = new BufferedWriter(new FileWriter(saveLocation, true));
+                for (Task task : taskList.getTasks()) {
+                    writer.append(task.saveString()).append("\n");
+                }
+                writer.close();
+            }
+        } catch (IOException e) {
+            ui.showSaveTasksError(e);
+        }
+    }
+
+    /**
+     * Loads saved tasks from the file and returns them as a list.
+     *
+     * @return A List of loaded Task objects.
+     * @throws DukeLoadTasksException If an error occurs while loading tasks.
+     * @throws DukeNoExistingTasksException If the save file does not exist.
+     */
+    public List<Task> loadSavedTasks() throws DukeLoadTasksException {
+        File saveFile = new File(saveLocation);
+        if (saveFile.exists()) {
+            try {
+                BufferedReader reader = new BufferedReader(new FileReader(saveLocation));
+                String line;
+                List<Task> tempTaskList = new ArrayList<>();
+                while ((line = reader.readLine()) != null) {
+                    String[] datum = line.split("\\|");
+                    try {
+                        switch (datum[0]) {
```

Comment:
I think the case statements are supposed to be on the same indentation level as switch according to the coding standard.